tests/dash/packets.py

```diff
@@ -0,0 +1,93 @@
+from ipaddress import ip_address
+
+import ptf.packet as scapy
+from ptf.mask import Mask
+import ptf.testutils as testutils
+
+from constants import *  # noqa: F403
+
+
+def inbound_vnet_packets(dash_config_info):
+    inner_packet = testutils.simple_udp_packet(
+        eth_src=dash_config_info[REMOTE_ENI_MAC],
+        eth_dst=dash_config_info[LOCAL_ENI_MAC],
+        ip_src=dash_config_info[REMOTE_CA_IP],
+        ip_dst=dash_config_info[LOCAL_CA_IP],
+    )
+    pa_match_vxlan_packet = testutils.simple_vxlan_packet(
+        eth_src=dash_config_info[REMOTE_PTF_MAC],
+        eth_dst=dash_config_info[DUT_MAC],
+        ip_src=dash_config_info[REMOTE_PA_IP],
+        ip_dst=dash_config_info[LOOPBACK_IP],
+        vxlan_vni=dash_config_info[VNET2_VNI],
+        ip_ttl=64,
+        inner_frame=inner_packet
+    )
+    expected_packet = testutils.simple_vxlan_packet(
+        eth_src=dash_config_info[DUT_MAC],
+        eth_dst=dash_config_info[LOCAL_PTF_MAC],
+        ip_src=dash_config_info[LOOPBACK_IP],
+        ip_dst=dash_config_info[LOCAL_PA_IP],
+        vxlan_vni=dash_config_info[VM_VNI],
+        ip_ttl=255,
+        ip_id=0,
+        inner_frame=inner_packet
+    )
+
+    pa_mismatch_vxlan_packet = pa_match_vxlan_packet.copy()
+    remote_pa_ip = ip_address(dash_config_info[REMOTE_PA_IP])
+    pa_mismatch_vxlan_packet["IP"].src = str(remote_pa_ip + 1)
+
+    masked_exp_packet = Mask(expected_packet)
+    masked_exp_packet.set_do_not_care_scapy(scapy.IP, "id")
+    masked_exp_packet.set_do_not_care_scapy(scapy.IP, "chksum")
+    masked_exp_packet.set_do_not_care_scapy(scapy.UDP, "sport")
+    masked_exp_packet.set_do_not_care_scapy(scapy.UDP, "chksum")
+
+    return inner_packet, pa_match_vxlan_packet, pa_mismatch_vxlan_packet, masked_exp_packet
+
+
+def outbound_vnet_packets(dash_config_info, inner_extra_conf={}):
+    if "proto" in inner_extra_conf:
+        proto = int(inner_extra_conf["proto"])
+        del inner_extra_conf["proto"]
+
+    inner_packet = testutils.simple_udp_packet(
+        eth_src=dash_config_info[LOCAL_ENI_MAC],
+        eth_dst=dash_config_info[REMOTE_ENI_MAC],
+        ip_src=dash_config_info[LOCAL_CA_IP],
+        ip_dst=dash_config_info[REMOTE_CA_IP],
+        **inner_extra_conf
+    )
+
+    if "proto" in inner_extra_conf:
+        inner_packet[scapy.IP].proto = proto
```

Comment:
The key "proto" is deleted in line 53, so the inner proto will not be changed here.